src/openlcb/BroadcastTimeDefs.cxx

```diff
@@ -189,20 +309,77 @@ int16_t BroadcastTimeDefs::string_to_rate_quarters(const std::string &srate)
 bool BroadcastTimeDefs::string_to_date(
     const std::string &sdate, int *year, int *month, int *day)
 {
+    // Note: The obvious way to simplify this method would be to make use of
+    //       strptime(). However, avoiding strptime() saves nearly 2K of code
+    //       space (ARMv7m, -Os).
+
+    size_t m_offset = string_find_first_not_space(sdate);
+    size_t d_offset = string_find_first_not_space(sdate, m_offset + 3);
+    size_t y_offset = sdate.find(',', d_offset + 1);
+    if (y_offset == std::string::npos)
+    {
+        // Invalid format, no comma.
+        return false;
+    }
+
+    y_offset = string_find_first_not_space(sdate, y_offset + 1);
+    if (y_offset == std::string::npos)
+    {
+        // Invalid format, no year.
+        return false;
+    }
+
     struct tm tm;
     memset(&tm, 0, sizeof(tm));
-    if (strptime(sdate.c_str(), "%b %e, %Y", &tm) == nullptr)
+
+    // Month.
     {
-        return false;
+        // Get a sub-string that matches the Mmm case format.
+        std::string m_str = sdate.substr(m_offset, 3);
```

Comment:
if m_str.size() < 3 return false;

after this the casing is a lot easier to read and a lot less code to run:
m_str[0] = toupper(m_str[0]);